Algorithms/Dynamic-Programming/Kadane Algorithm.cpp

```diff
@@ -0,0 +1,103 @@
+#include<iostream>
+#include<vector>
+
+using namespace std;
+
+vector<int> maxSubArray(const vector<int> &A)
+{
+    int maxx=INT_MIN,sum=0,i,l=0,r=0,dif=INT_MAX,low_index;
+    if(A.size()==1)
+        maxx=A[0];
+    else
+    {
+        int n=A.size();
+        int c=0;
+        //check if all elements are negative
+        for(i=0;i<n;i++)
+        {
+            if(A[i]>0)break;
+            else c++;
+        }
+        //if all elements are negative,maximum sum is the maximum number in the array
```

Comment:
Can you handle both the cases via a general **Dynamic Programming** approach considering `dp[i]` to be the maximum sum ending at `a[i]` ? Since the approach you've used currently, is just making a lot of cases. 

If yes, update the current solution to a DP solution and elaborate on the time complexity then.

Reply:
okay trying to make these changes